proof/refine/ARM/TcbAcc_R.thy

```diff
@@ -2367,10 +2367,22 @@ lemma get_sc_refill_ready_sp:
            \<and> P s\<rbrace>"
   by (wpsimp simp: obj_at_def)
 
-lemma rescheduleRequired_corres:
-  "corres dc (valid_tcbs and weak_valid_sched_action and pspace_aligned and pspace_distinct)
+definition weaker_valid_sched_action where
+  "weaker_valid_sched_action s \<equiv>
+   \<forall>t. scheduler_action s = switch_thread t \<longrightarrow>
+       tcb_at t s \<and> (bound_sc_tcb_at ((\<noteq>) None) t s \<longrightarrow> released_sc_tcb_at t s)"
+
+lemma weak_valid_sched_action_strg:
+  "weak_valid_sched_action s \<longrightarrow> weaker_valid_sched_action s"
+  by (fastforce simp: weak_valid_sched_action_def weaker_valid_sched_action_def
```

Comment:
Maybe `vs_all_heap_simps` or `obj_at_kh_kheap_simps` or both would do the trick here, or at least reduce the number of things you have to give to simp